kore/src/Kore/Validate/PatternVerifier.hs

```diff
@@ -230,15 +232,39 @@ verifyOperands operandSort = \operator -> do
     traverse verifyChildWithSort operator
 {-# INLINE verifyOperands #-}
 
+internalizeAnd ::
+    And Sort Verified.Pattern ->
+    BinaryAnd Sort Verified.Pattern
+internalizeAnd And{andSort, andChildren = [andFirst, andSecond]} =
+    BinaryAnd{andSort, andFirst, andSecond}
+internalizeAnd And{andSort, andChildren} =
+    let first = head andChildren
+     in let second = internalizeAnd And{andSort, andChildren = tail andChildren}
+         in BinaryAnd{andSort, andFirst = first, andSecond = synthesize (Internal.AndF second)}
```

Comment:
I understand that the case of getting an `And` with 0 or 1 child is unlikely, but I think it would be better to handle this case with an explicit call to `error`, i.e. add another equation
```suggestion
internalizeAnd ::
    And Sort Verified.Pattern ->
    BinaryAnd Sort Verified.Pattern
internalizeAnd And{andSort, andChildren}
    | length andChildren <= 1 = error "Cannot internalize And with less than two children"
    | otherwise = case andChildren of
        [andFirst, andSecond] -> BinaryAnd{andSort, andFirst, andSecond}
        _ ->
            let first = head andChildren
                second = internalizeAnd And{andSort, andChildren = tail andChildren}
             in BinaryAnd{andSort, andFirst = first, andSecond = synthesize (Internal.AndF second)}
```

Reply:
definitely agree. using head/tail is usually not a good idea because when the list is empty you get a generic error and don't know where it's coming from. @geo2a's suggestion throws a better error message.